Gems/Atom/Tools/MaterialCanvas/Code/Source/Document/MaterialGraphCompiler.cpp

```diff
@@ -1122,37 +1176,80 @@ namespace MaterialCanvas
             }
         }
 
-        CompileGraphCompleted();
+        m_compileGraph = false;
+        m_compileAssets = !m_generatedFiles.empty();
+        m_compileAssetIndex = 0;
         return true;
     }
 
-    void MaterialGraphCompiler::CompileGraphStarted() const
+    void MaterialGraphCompiler::CompileGraphStarted()
     {
-        m_compileGraphQueued = false;
+        m_compileGraph = false;
+        m_compileAssets = false;
+        m_compileAssetIndex = 0;
         m_slotValueTable.clear();
         m_generatedFiles.clear();
-        AZ_TracePrintf_IfTrue("MaterialGraphCompiler", IsCompileLoggingEnabled(), "Compile graph started.\n");
-        MaterialGraphCompilerNotificationBus::Event(m_toolId, &MaterialGraphCompilerNotificationBus::Events::OnCompileGraphStarted, m_documentId);
+        AZ_TracePrintf_IfTrue("MaterialGraphCompiler", IsCompileLoggingEnabled(), "Compiling started.\n");
+
+        AtomToolsFramework::AtomToolsMainWindowRequestBus::Event(
+            m_toolId,
+            &AtomToolsFramework::AtomToolsMainWindowRequestBus::Events::SetStatusMessage,
+            AZStd::string::format("Compiling %s (Started)", GetGraphPath().c_str()));
+
+        MaterialGraphCompilerNotificationBus::Event(
+            m_toolId, &MaterialGraphCompilerNotificationBus::Events::OnCompileGraphStarted, m_documentId);
     }
 
-    void MaterialGraphCompiler::CompileGraphFailed() const
+    void MaterialGraphCompiler::CompileGraphFailed()
     {
-        m_compileGraphQueued = false;
+        m_compileGraph = false;
+        m_compileAssets = false;
+        m_compileAssetIndex = 0;
         m_slotValueTable.clear();
         m_generatedFiles.clear();
-        AZ_TracePrintf_IfTrue("MaterialGraphCompiler", IsCompileLoggingEnabled(), "Compile graph failed.\n");
-        MaterialGraphCompilerNotificationBus::Event(m_toolId, &MaterialGraphCompilerNotificationBus::Events::OnCompileGraphFailed, m_documentId);
+        AZ_TracePrintf_IfTrue("MaterialGraphCompiler", IsCompileLoggingEnabled(), "Compiling failed.\n");
+
+        AtomToolsFramework::AtomToolsMainWindowRequestBus::Event(
+            m_toolId,
+            &AtomToolsFramework::AtomToolsMainWindowRequestBus::Events::SetStatusError,
+            AZStd::string::format("Compiling %s (Failed)", GetGraphPath().c_str()));
```

Comment:
Wherever possible, I think I'd rather have the same message be used in both the status bar and in the log. That way if I see a message in the status bar I can search the log for that message, and see if there is other relevant information around that event.